x/interchainstaking/keeper/ibc_packet_handlers_test.go

```diff
@@ -771,6 +773,168 @@ func (suite *KeeperTestSuite) TestHandleWithdrawForUserLSM() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestHandleWithdrawRewards() {
+	val := addressutils.GenerateValAddressForTest().String()
+	user := addressutils.GenerateAddressForTestWithPrefix("quick")
+	tests := []struct {
+		name      string
+		setup     func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone)
+		msg       func(zone *icstypes.Zone) sdk.Msg
+		checks    func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone)
+		triggered bool
+		err       bool
+	}{
+		{
+			name:   "wrong msg",
+			setup:  func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawValidatorCommission{
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		{
+			name: "wrong context",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.ConnectionId = ""
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		// at this time, the waitgroup will be uint32_max so it's not gonna trigger redemption rate
+		{
+			name: "negative waitgroup",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.WithdrawalWaitgroup = 0
```

Comment:
I see we don't have a check whether `zone.WithdrawalWaitgroup < 0` or not, I think we should add it maybe in case the line below cc @joe-bowman  https://github.com/quicksilver-zone/quicksilver/blob/543f819670378f1be207be43054514583c7de99d/x/interchainstaking/keeper/ibc_packet_handlers.go#L1132